django/thunderstore/frontend/api/experimental/views/community_package_list.py

```diff
@@ -0,0 +1,253 @@
+from django.db.models import Count, Prefetch, Q, QuerySet, Sum
+from django.http import HttpRequest, HttpResponse
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import status
+from rest_framework.generics import get_object_or_404
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from thunderstore.cache.cache import CacheBustCondition
+from thunderstore.cache.pagination import CachedPaginator
+from thunderstore.community.models import (
+    Community,
+    PackageListingReviewStatus,
+    PackageListingSection,
+)
+from thunderstore.frontend.api.experimental.serializers.views import (
+    CommunityPackageListSerializer,
+    PackageCategorySerializer,
+)
+from thunderstore.repository.models import Package
+
+
+class CommunityPackageListApiView(APIView):
+    """
+    Return paginated list of community's packages.
+    """
+
+    permission_classes = []
+
+    @swagger_auto_schema(
+        responses={200: CommunityPackageListSerializer()},
+        operation_id="experimental.frontend.community.packages",
+    )
+    def get(self, request: HttpRequest, **kwargs) -> HttpResponse:
+        communities = Community.objects.prefetch_related("package_categories")
+        community = get_object_or_404(communities, identifier=kwargs.get("community"))
+        package_qs = self.get_queryset(community)
+        package_qs = self.filter_deprecated(request, package_qs)
+        package_qs = self.filter_nsfw(request, package_qs)
+        package_qs = self.filter_by_categories(request, package_qs)
+        package_qs = self.filter_by_section(request, package_qs)
+        package_qs = self.filter_by_query(request, package_qs)
+        package_qs = self.order_queryset(request, package_qs)
+        package_qs = self.paginate(request, package_qs)
+        serializer = self.serialize_results(community, package_qs)
+
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    def get_full_cache_vary(self, request: HttpRequest) -> str:
+        """
+        Return cache key for CachedPaginator.
+        """
+        cache_vary = self.kwargs["community"]
+        cache_vary += f".{'deprecated' in request.query_params}"
```

Comment:
Well I guess if we want to change it, I might as well do the changes now.

Reply:
Well, it seems the old, non-experimental implementation is no better, since `QueryDict` returns all values as strings (unless I'm missing something), so the only falsy values are a missing key and an empty string. So even if `&deprecated=0` or `&deprecated=false` is sent, it evaluates to `True`.

I can check for empty string (in fact this I've already implemented), but trying to cast other values to correct types seems like a fool's errand. So if we want to have some falsy value, I'd say we should agree on one, and then use it in requests and check for it - and it only - on backend. Thoughts?

Reply:
```
qp = PackageSearchQueryParameterSerializer(data=request.query_params)
if not qp.is_valid():
    return Response(qp.errors, status=status.HTTP_400_BAD_REQUEST)

params: OrderedDict = qp.validated_data
```

Something like this seems to work so I'll go with it. 👍🏻 for the idea.